concepts/functions/introduction.md

```diff
@@ -1,17 +1,102 @@
 # Introduction
 
-A [`function`][function] is a named block of code that takes input and produces output when invoked. Functions are used to perform specific and repetitive tasks throughout a program.
+A [`function`][function] is a way to compartmentalize code so it can be called from one or more places in the program.
+Functions are used to perform specific and repetitive tasks throughout a program.
 
-More formally:  a function is any Python object to which the [`function call`][calls] operation can be applied. The body of the function contains one or more statements which will return some value to the calling code/caller. The function object can define/receive zero or more parameters/arguments (_including other functions_).
+More formally: a function is any Python object to which the [`function call`][calls] operation can be applied.
+A function may be used to [`return`][return] one or more values as a result of some operation(s), or it may be used for one or more [`side effects`][side effects].
+If a function does not specify a return value it will still return `None`. 
 
-[Function definitions][function-definitions] (`def <function_name>:`) form an _executable statement_ which binds the `<function_name>` to the function object definition (_the wrapper around the executable statements in the function body_) in the current local namespace.
+Following is an example of a function with a side effect
 
-Functions are [_first class objects_][first-class objects], which means they can be assigned to variables, passed as arguments to other functions, used as a `return` value, and wrapped around/nested to form decorators and closures.
+```python
+def hello():
+    print("Hello")
 
-Python also provides many top-level [_built-in functions_][built-in functions] for common operations.
 
+hello()
+
+```
+
+The [`def`][def] keyword is used to signal the beginning of the function definition.
+`hello` is the name of the function.
+The empty parentheses represent that no values are being passed to the function.
+The body of the function is the single `print("Hello")` statement.
+`print` is also a function.
+The `"Hello"`  in the `print` function's parentheses is known as an [`argument`][arguments].
+The argument is used by the `print` function to know what to print. 
+Since the `hello` function does not specify a `return` value, it executes its side effect - which is calling the `print` function - and returns `None`.
```

Comment:
There is a link to side effects on line 7.

A function may be used to [`return`][return] one or more values as a result of some operation(s), or it may be used for one or more [`side effects`][side effects].

Although I would be happier with a different article to link to. Okay, I think I found one...

Reply:
But it's on Medium, so a limited viewing before paywall, which is why I chose the worse one earlier.